x/gamm/pool-models/stableswap/pool_test.go

```diff
@@ -1068,3 +1067,140 @@ func TestValidatePoolLiquidity(t *testing.T) {
+			expectPass:     true,
+		},
+		"uneven two-asset pool": {
+			baseDenom:      "foo",
+			quoteDenom:     "bar",
+			poolAssets:     twoUnevenStablePoolAssets,
+			scalingFactors: defaultTwoAssetScalingFactors,
+			expectPass:     true,
+		},
+		"even three-asset pool": {
+			baseDenom:      "asset/a",
+			quoteDenom:     "asset/b",
+			poolAssets:     threeEvenStablePoolAssets,
+			scalingFactors: defaultThreeAssetScalingFactors,
+			expectPass:     true,
+		},
+		"even three-asset pool with large scaling factors": {
+			baseDenom:      "asset/a",
+			quoteDenom:     "asset/b",
+			poolAssets:     threeEvenStablePoolAssets,
+			scalingFactors: []uint64{10000, 10000, 10000},
+			expectPass:     true,
+		},
+		"uneven three-asset pool (a -> b)": {
+			baseDenom:  "asset/a",
+			quoteDenom: "asset/b",
+			poolAssets: sdk.NewCoins(
+				sdk.NewInt64Coin("asset/a", 10_000_000_000),
+				sdk.NewInt64Coin("asset/b", 20_000_000_000),
+				sdk.NewInt64Coin("asset/c", 30_000_000_000),
+			),
+			scalingFactors: defaultThreeAssetScalingFactors,
+			expectPass:     true,
+		},
+		"uneven three-asset pool (b -> a)": {
+			baseDenom:  "asset/b",
+			quoteDenom: "asset/a",
+			poolAssets: sdk.NewCoins(
+				sdk.NewInt64Coin("asset/a", 10_000_000_000),
+				sdk.NewInt64Coin("asset/b", 20_000_000_000),
+				sdk.NewInt64Coin("asset/c", 30_000_000_000),
+			),
+			scalingFactors: defaultThreeAssetScalingFactors,
+			expectPass:     true,
+		},
+		"uneven three-asset pool large scaling factors (a -> b)": {
+			baseDenom:  "asset/a",
+			quoteDenom: "asset/b",
+			poolAssets: sdk.NewCoins(
+				sdk.NewInt64Coin("asset/a", 10_000_000_000),
+				sdk.NewInt64Coin("asset/b", 20_000_000_000),
+				sdk.NewInt64Coin("asset/c", 30_000_000_000),
+			),
+			scalingFactors: []uint64{10000, 10000, 10000},
+			expectPass:     true,
+		},
+		"uneven three-asset pool large scaling factors (b -> a)": {
+			baseDenom:  "asset/b",
+			quoteDenom: "asset/a",
+			poolAssets: sdk.NewCoins(
+				sdk.NewInt64Coin("asset/a", 10_000_000_000),
+				sdk.NewInt64Coin("asset/b", 20_000_000_000),
+				sdk.NewInt64Coin("asset/c", 30_000_000_000),
+			),
+			scalingFactors: []uint64{10000, 10000, 10000},
+			expectPass:     true,
+		},
+		"uneven 3-asset pool with uneven scaling factors (a -> b)": {
+			baseDenom:  "asset/a",
+			quoteDenom: "asset/b",
+			poolAssets: sdk.NewCoins(
+				sdk.NewInt64Coin("asset/a", 12_345_678_910),
+				sdk.NewInt64Coin("asset/b", 10_987_654_321),
+				sdk.NewInt64Coin("asset/c", 8_452_398_713),
+			),
+			scalingFactors: []uint64{36, 578, 253},
+			expectPass:     true,
+		},
+		"uneven 3-asset pool with uneven scaling factors (b -> a)": {
+			baseDenom:  "asset/a",
+			quoteDenom: "asset/b",
+			poolAssets: sdk.NewCoins(
+				sdk.NewInt64Coin("asset/a", 12_345_678_910),
+				sdk.NewInt64Coin("asset/b", 10_987_654_321),
+				sdk.NewInt64Coin("asset/c", 8_452_398_713),
+			),
+			scalingFactors: []uint64{36, 578, 253},
+			expectPass:     true,
+		},
```

Comment:
Need to add a test case for even two asset pool & three asset pool, with diff scaling factors